fdbserver/storageserver.actor.cpp

```diff
@@ -5140,9 +5167,10 @@ ACTOR Future<Void> getMappedKeyValuesQ(StorageServer* data, GetMappedKeyValuesRe
 			try {
 				// Map the scanned range to another list of keys and look up.
 				GetMappedKeyValuesReply _r =
-				    wait(mapKeyValues(data, getKeyValuesReply, req.mapper, &req, tenantPrefix, req.matchIndex));
+				    wait(mapKeyValues(data, getKeyValuesReply, req.mapper, &req, req.matchIndex, &remainingLimitBytes));
 				r = _r;
 			} catch (Error& e) {
+				// TODO: cannot allow txn_too_old to reach here, need to handle it gracefully, add test
```

Comment:
I think `transaction_too_old` here actually is fine. It'll get propagated up to the user, and it would be (I think) the expected behavior if doing a read-range close to the 5-second limit boundary. That being said, it would be kind of nice if it could return whatever it had already read rather than just throwing an error, but probably not the end of the world,

Reply:
thanks for explaining and makes total sense, it is something I want to get to the bottom to, before merging the PR.